### ISSUE TITLE
How does XGBoost control overfitting

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#84]
There are in general two ways that you can control overfitting in XGBoost:

- The first way is to directly control model complexity.

This includes max_depth, <code>min_child_weight</code> and gamma.

- The second way is to add randomness to make training robust to noise.

This includes subsample and colsample_bytree.

You can also reduce stepsize eta. Remember to increase num_round when you do so.

**Comment by WatershedXiaolan** [comment]
max_depth [default=6]

Maximum depth of a tree. Increasing this value will make the model more complex and more likely to overfit. 0 is only accepted in lossguided growing policy when tree_method is set as hist and it indicates no limit on depth. Beware that XGBoost aggressively consumes memory when training a deep tree.

range: [0,∞] (0 is only accepted in lossguided growing policy when tree_method is set as hist)

**Comment by WatershedXiaolan** [comment]
min_child_weight

**Comment by WatershedXiaolan** [comment]
gamma [default=0, alias: min_split_loss]

Minimum loss reduction required to make a further partition on a leaf node of the tree. The larger gamma is, the more conservative the algorithm will be.

range: [0,∞]

**Comment by WatershedXiaolan** [comment]
subsample [default=1]

Subsample ratio of the training instances. Setting it to 0.5 means that XGBoost would randomly sample half of the training data prior to growing trees. and this will prevent overfitting. Subsampling will occur once in every boosting iteration.

range: (0,1]

**Comment by WatershedXiaolan** [comment]
colsample_bytree is the subsample ratio of columns when constructing each tree. Subsampling occurs once for every tree constructed.
